### ISSUE TITLE
Lohmann 2M cirrus nucleation (nic_cirrus=1): ICNC gradient is NaN for ice radius below ~3e-7 m, and the forward candidate is unphysical there

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#846]
## What is wrong

`jcm/physics/clouds/lohmann_2m/assembly.py:449`, the default `nic_cirrus=1` branch:

```python
icnc_candidate = 0.75 / (pi * params.rhoice) * air_density * cloud_ice_in_cloud / jnp.maximum(ice_radius_mean**3, params.d_epsilon)
```

Both AD modes weight the denominator by `den**-2`, which float32 evaluates as `1/(den*den)` and returns `inf` once `ice_radius_mean**3` is below ~2**-63, i.e. `ice_radius_mean` below ~3e-7 m. On a cell where `ll2_ic` holds (cold, cloudy, ICNC at or below `icemin` — exactly the nucleating-cirrus cell this branch exists for) the ICNC output's jvp is `nan` **even for a zero tangent**, off a finite forward pass. Verified at `r` in {0, 1e-7, 3e-7} m; clean at 1e-6 m and above. Found by the #820 sweep (the same float32 `den**-2` mechanism fixed in `echam_1m.py::_saturation_adjustment_layer` and `grey_two_stream/planck.py` in that PR).

## Why it is not a guard fix

Raising the floor to the float32-differentiable bound (`_MIN_DIFFERENTIABLE_DENOMINATOR = 1e-18`, see `echam_1m.py`) would make the derivative finite, but the forward value on that branch is already unphysical: `icnc_candidate` is 1.2e11 m^-3 at `r = 1e-7` m and 1.2e20 at `r = 0`, against realistic cirrus ICNC of 1e3 to 1e6 m^-3. A physically motivated minimum ice radius (or a cap on the candidate) is a science choice against the ECHAM-HAM reference (`mo_cloud_micro_2m.f90`, the `nic_cirrus` branch) and needs a `docs/source/science/` update, so it is not folded into the numerics-only #820 fixes.

## Where the hooks are

- `assembly.py:440-449` (`ll2_ic`, the `nic_cirrus == 1` candidate); `params.d_epsilon`, `params.icemin`.
- `jcm/physics/clouds/lohmann_2m_test.py` — the scheme-level `check_gradients` fences added in #820 to extend with a nucleating-cirrus column once the floor is decided.
- Fixed instance of the same mechanism to copy: `echam_1m.py::_saturation_adjustment_layer` and its `_MIN_DIFFERENTIABLE_DENOMINATOR` rationale.
